### PR TITLE
Set permissions on the Grafana Agent [Flow] folder...

### LINKED ISSUE #6522: Ensure ACLs are set correctly when using non-default install location on windows.
### What's wrong?

Its possible that a non-default install location for windows can have defaults that allow an attacker to change the executable or add dlls if not properly set by the user.

The agent doesnt directly load any dlls, but its possible dependencies do. 

### Steps to reproduce

Use non default install and don't change permissions

### System information

windows

### Software version

_No response_

### Configuration

_No response_

### Logs

_No response_

### PR DESCRIPTION
**Pull request by erikbaranowski**
when installing via the windows installer rather than relying on the parent folder permissions.

#### Which issue(s) this PR fixes

Fixes #6522

#### PR Checklist

- [x] CHANGELOG.md updated
- [ ] Documentation added
- [ ] Tests updated
- [ ] Config converters updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3878,7 +3882,7 @@ v0.1.0 (2020-03-16)
 
 > First release!
 
-### Features
+### Featuresgit status
```
Comment (erikbaranowski) [inline CHANGELOG.md:3885]:
```suggestion
### Features
```

File: tools/make/packaging.mk [hunk 2]
```diff
@@ -388,8 +388,9 @@ ifeq ($(USE_CONTAINER),1)
 else
 	cp ./dist/grafana-agent-windows-amd64.exe ./packaging/grafana-agent/windows
 	cp LICENSE ./packaging/grafana-agent/windows
-	# quotes around mkdir are manadory. ref: https://github.com/grafana/agent/pull/5664#discussion_r1378796371
+	# quotes around mkdir are mandatory. ref: https://github.com/grafana/agent/pull/5664#discussion_r1378796371
 	"mkdir" -p dist
+	curl -O https://nsis.sourceforge.io/mediawiki/images/4/4a/AccessControl.zip && unzip AccessControl.zip -d /usr/share/nsis/ && cp /usr/share/nsis/Plugins/i386-unicode/AccessControl.dll /usr/share/nsis/Plugins/x86-unicode/
```
Comment (erikbaranowski) [inline tools/make/packaging.mk:393]: This can probably be done in the build image instead.

Comment (mattdurham) [reply tools/make/packaging.mk:393]: This feels semi reasonable :)

File: packaging/grafana-agent-flow/windows/install_script.nsis [hunk 3]
```diff
@@ -164,6 +166,19 @@ Function InitializeRegistry
...
 
+Function SetFolderPermissions
+    # Set permissions on the install directory
+    SetOutPath $INSTDIR
+    AccessControl::DisableFileInheritance $INSTDIR
+    AccessControl::SetFileOwner $INSTDIR "Administrators"
+    AccessControl::ClearOnFile  $INSTDIR "Administrators" "FullAccess"
+    AccessControl::SetOnFile    $INSTDIR "SYSTEM" "FullAccess"
+    AccessControl::GrantOnFile  $INSTDIR "Everyone" "ListDirectory"
+    AccessControl::GrantOnFile  $INSTDIR "Everyone" "GenericExecute"
```
Comment (mattdurham) [inline packaging/grafana-agent-flow/windows/install_script.nsis:177]: GenericExecute sounds scary?

Comment (mattdurham) [reply packaging/grafana-agent-flow/windows/install_script.nsis:177]: I also hate that GrantOnFile is both file and directory, made me look up the docs on this.

**Review by mattdurham — APPROVED** [review]
Tested and looks good

**Review by mattdurham — APPROVED** [review]
Lgtm

**Pull request merged**